### PR TITLE
Issue #12974: Added test cases for fallthrough doc examples

### PR DESCRIPTION
**Pull request by kalpadiptyaroy**
Issue #12974: Added test cases for fallthrough doc examples.

### CODE REVIEW

File: config/checkstyle-examples-suppressions.xml [hunk 1]
```diff
@@ -49,4 +49,8 @@
   <suppress checks="FileLength"
     files="visibilitymodifier[\\/]Example\d+.java"/>
 
+  <!-- Examples for understanding fallthrough cases -->
+  <suppress checks="FallThrough"
+    files="fallthrough[\\/]Example\d+.java"/>
+
```
Comment (romani) [inline config/checkstyle-examples-suppressions.xml:55]: Are you sure that we need this ?
Examples are all about to make code bad on purpose, we should not even validate them at all.
Try remove, if there is violation, please share it.

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: config/checkstyle-examples-suppressions.xml (hunk 1 above)
Comment (kalpadiptyaroy) [reply config/checkstyle-examples-suppressions.xml:55]: ok. working on it

Comment (kalpadiptyaroy) [reply config/checkstyle-examples-suppressions.xml]: This is done.

**Comment by kalpadiptyaroy** [comment]
@romani. But now there is another problem. Checkstyle expects the line length to be 85 characters max. But, the violation messages of the fall thorugh examples are exceeding the limit and hence I am getting this error.
```
[ERROR] [checkstyle] [ERROR] /root/project/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/fallthrough/Example1.java:21: Line is longer than 85 characters (found 89). [LineLength]
[ERROR] [checkstyle] [ERROR] /root/project/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/fallthrough/Example1.java:32: Line is longer than 85 characters (found 89). [LineLength]
[ERROR] [checkstyle] [ERROR] /root/project/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/fallthrough/Example2.java:23: Line is longer than 85 characters (found 89). [LineLength]
[ERROR] [checkstyle] [ERROR] /root/project/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/fallthrough/Example2.java:34: Line is longer than 85 characters (found 89). [LineLength]
[ERROR] [checkstyle] [ERROR] /root/project/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/fallthrough/Example2.java:38: Line is longer than 85 characters (found 90). [LineLength]
[ERROR] [checkstyle] [ERROR] /root/project/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/fallthrough/Example3.java:23: Line is longer than 85 characters (found 89). [LineLength]
```
I have tried shortening them with `.* ` regex expression but then also getting `ForbiddenRegexExpression` error.
Any clues? How to proceed?

**Comment by romani** [comment]
I did update like:
```
diff --git a/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/fallthrough/Example3.java b/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/fallthrough/Example3.java
index 42db2ef..f5b6122 100644
--- a/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/fallthrough/Example3.java
+++ b/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/fallthrough/Example3.java
@@ -20,7 +20,7 @@ class Example3 {
       switch (i) {
         case 1:
           i++;
-        case 2: // violation 'Fall through from previous branch of the switch statement.'
+        case 2: // violation 'Fall through from previous branch of the switch'
           i++;
           break;
         case 3:

```

it worked.

**Comment by romani** [comment]
@stoyanK7 , fyi, we need to consider experience of this PR to let us create wiki page on how to do example files and resovle common problems with them.

**Review by romani — APPROVED** [review]
ok to merge if CI pass

**Comment by kalpadiptyaroy** [comment]
> I did update like:
> 
> ```
> diff --git a/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/fallthrough/Example3.java b/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/fallthrough/Example3.java
> index 42db2ef..f5b6122 100644
> --- a/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/fallthrough/Example3.java
> +++ b/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/coding/fallthrough/Example3.java
> @@ -20,7 +20,7 @@ class Example3 {
>        switch (i) {
>          case 1:
>            i++;
> -        case 2: // violation 'Fall through from previous branch of the switch statement.'
> +        case 2: // violation 'Fall through from previous branch of the switch'
>            i++;
>            break;
>          case 3:
> ```
> 
> it worked. I pushed changes for all violations.

That's good. But how did it worked without a regex expression?

**Comment by romani** [comment]
It is regexp, even no regexp special symbols are in string, play with it at https://regex101.com/

**Comment by kalpadiptyaroy** [comment]
> It is regexp, even no regexp special symbols are in string, play with it at https://regex101.com/

Cool got it.

**Comment by romani** [comment]
@kalpadiptyaroy , please read inspection violations 
And fix them https://app.circleci.com/pipelines/github/checkstyle/checkstyle/20161/workflows/151845fd-5752-4be8-9a0a-c8d92c722ec3/jobs/358754/artifacts

RedundantMethodOverride.xml must be fixed, 
YAMLSchemaValidation.xml I don't think are related, might be good after restart of inspection.

**Review by nrmancuso — APPROVED** [review]

**Comment by nrmancuso** [comment]
Ok to merge once CI is happy.

**Comment by romani** [comment]
@kalpadiptyaroy , CI should be green to merge PR.

**Comment by kalpadiptyaroy** [comment]
> @kalpadiptyaroy , CI should be green to merge PR.

@romani. CI is green and happy. We are good to go ahead. Please merge the branch.

**Comment by nrmancuso** [comment]
@kalpadiptyaroy there should only be one commit in this PR, please squash commits

**Comment by kalpadiptyaroy** [comment]
> @kalpadiptyaroy there should only be one commit in this PR, please squash commits

Done.

**Pull request merged**